cli/internal/core/engine.go

```diff
@@ -158,6 +158,12 @@ func (e *Engine) Prepare(options *EngineBuildingOptions) error {
 
 	traversalQueue := []string{}
 
+	// make a map of taskNames passed in
+	missing := map[string]bool{}
+	for _, taskName := range taskNames {
+		missing[taskName] = true
+	}
```

Comment:
Let's use our Set implementation rather than rolling one inline.

```suggestion
	missing := util.SetFromStrings(taskNames)
```